coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -975,4 +1001,18 @@ static StructTypeInfo structTypeInfoFromCols(List<FieldSchema> cols) {
         cols.stream().map(f -> TypeInfoUtils.getTypeInfoFromTypeString(f.getType())).collect(Collectors.toList());
     return (StructTypeInfo) TypeInfoFactory.getStructTypeInfo(fieldNames, fieldTypeInfos);
   }
+
+  /**
+   * Reorders an option schema so that the type of the provided default value is the first type in the option schema
+   *
+   * e.g. If the schema is [null, int] and the default value is 1, the returned schema is [int, null]
+   * If the schema is not an option schema or if there is no default value, schema is returned as-is
+   */
+  static Schema reorderOptionIfRequired(Schema schema, Object defaultValue) {
+    if (isNullableType(schema) && defaultValue != null && schema.getTypes().get(0).getType() == Schema.Type.NULL) {
```

Comment:
should we also check if `defaultValue.toString != null`?

Reply:
If `defaultValue` is not `null`, I think `defaultValue.toString != null` is always true?

Reply:
(obj = null) is not a safe null check. https://stackoverflow.com/questions/2369226/object-null-or-null-object mentions that if obj datatype is Boolean, this check could be problematic. 
Do we use bool/boolean? 

Its probably safe to replace with `Objects.toString(object, "isNull") != "isNull"` as mentioned https://stackoverflow.com/a/13684257 . What do you think?

Reply:
I still don't think it's necessary for this method, I have tried to reorder the following schema:
```
{
    "name" : "Boolean_field",
    "type" : ["boolean", "null"],
    "default": false
}
```
the default value's type is Boolean only if it's `true` or `false`, which can't be null. The current `reorderOptionIfRequired` method handles the above test case well.